Feature/freeze chrome driver and binary versions

bodyangug opened the pull request.

bodyangug committed:
feat: add tasks for freezing chromedriver and binary chrome files

bodyangug committed:
chore: move urls to property file

bodyangug committed:
fix: remove a clean task

RomanDavlyatshin commented on docker/scripts/create.bat:
```diff
@@ -0,0 +1,6 @@
+@echo off
```
Reason for creating new script file to launch demo (we already have `demo-up.sh`) 

Maybe adding `.bat` scripts is a good idea, since we cannot be sure if a user on windows has gitbash/WSL. But we'll have to create separate scripts for all actions - that's a separate PR for sure

bodyangug replied:
my bad, that file should not be here.

RomanDavlyatshin commented on gradle.properties:
```diff
@@ -0,0 +1,2 @@
+chromeDriverUrl=https://chromedriver.storage.googleapis.com/114.0.5735.90/chromedriver_win32.zip
```
Properties are added but never used

bodyangug replied:
pr was updated

RomanDavlyatshin commented on build.gradle:
```diff
@@ -34,3 +34,57 @@ allprojects {
 }
+
+// 0. clean up old files and register that task
+tasks.register('cleanUpChromeFolders', Delete) {
```
Creating separate task for each tiny action seems excessive.
Can't we just create 2 tasks?
```
task.register("downloadChromium") {
  // cleanup
  // ensure dir exists
  // download
  // unzip
}

task.register("downloadChromeDriver") {
  // cleanup
  // ensure dir exists
  // download
  // unzip
}
```

bodyangug replied:
done

RomanDavlyatshin commented on build.gradle:
```diff
@@ -34,3 +34,57 @@ allprojects {
+tasks.register('prepareDirs') {
+    mustRunAfter clean
+    doFirst {
+        mkdir "./chrome-binary/"
```
1. Add download folder(s) to `.gitignore`
2. It's a nitpicking, but just use the same folder

RomanDavlyatshin commented on build.gradle:
```diff
@@ -34,3 +34,57 @@ allprojects {
+}
+tasks.register('downloadChromeBinary', Exec) {
+    dependsOn('prepareDirs')
+    commandLine 'curl', '-L', '-o', './chrome-binary/chrome-win.zip',
```
1. We _have to_ use _Chromium_ and not Chrome because group policies may prevent users from installing/using older Chrome versions.
Chromium has no such limitations.
2. Refer to https://chromium.cypress.io/ for Chromium versions
3. Re-upload chrome binary to realworld demo repository to avoid headache in case website from point above is unavailable

bodyangug committed:
fix: remove a create script

bodyangug committed:
refactor: update .gitignore
- rename chrome-binary to chromium-binary
- rename downloadChromeBinary task to downloadChromiumBinary

bodyangug committed:
refactor: reduce the number of tasks

bodyangug committed:
refactor: move assemble dependsOn
